### PR TITLE
Add review documents

### PR DESCRIPTION
**Pull request by fmvilas**
### What?
It adds documents for the reviewing process.

### Why?
To help writing the review of a new AsyncAPI version.

### CODE REVIEW

File: documents/processes/review/EXAMPLE.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# AsyncAPI Review Document
+
+|Project name|Reviewers|Company|
+|------------|---------|-------|
+|Awesome project|Jack Johnson, John Jackson|ACME|
```
Comment (MikeRalphson) [inline documents/processes/review/EXAMPLE.md]: Should we capture the version of the spec. reviewed (i.e. if there are more than one RC / for future versions)?

File: documents/processes/review/EXAMPLE.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+### 1. The AMQP extension doesn't work for us.
+The AMQP extension doesn't allow me to define my queues and exchanges.
+
+#### We think it can be solved by...
+Improving the AMQP extension.
+
+#### Overall, we think this challenge/difficulty is...
+  * [ ] A "nice to have".
+  * [ ] A "must have" for the next minor version.
```
Comment (jhigginbotham) [inline documents/processes/review/EXAMPLE.md]: Wondering if a rephrase of the must-haves might help, e.g. "A blocker for us to adopt v2 of the spec"? The reviewer may not have a good idea of which version it belongs in, but for us we need to know if this blocks them from adopting or if they can proceed with a workaround.

Comment (fmvilas) [reply documents/processes/review/EXAMPLE.md]: Changed. Thanks for the feedback!

File: documents/processes/review/EXAMPLE.md (hunk 1 above)
Comment (fmvilas) [reply documents/processes/review/EXAMPLE.md]: Added. Thanks for the feedback!

**Comment by jhigginbotham** [comment]
LGTM

File: documents/processes/review/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Review Process
+
+This document describes the steps you should follow when reviewing a new version of AsyncAPI.
+
+## Steps
+
+1. Please, communicate the project director that you want to start the review process.
```
Comment (lifewingmate) [inline documents/processes/review/README.md]: 1. Replace current "Please, communicate"...to "Let the project director know that you want to start the review process."

Comment (fmvilas) [reply documents/processes/review/README.md]: Done. Thanks!

**Comment by derberg** [comment]
It is actually not included in the release. This was an unintentional release caused by the release automation initial run. 1.0 version of the spec was released a long time ago. The release mentioned by the bot was removed.

**Pull request merged**